### PR TITLE
Feat: update for NS7

### PR DESCRIPTION
**Pull request by bradmartin**
Implements NS7 support for the plugin source.

### CODE REVIEW

**Comment by bradmartin** [comment]
[nativescript-download-progress-2.0.0.tgz.zip](https://github.com/tobydeh/nativescript-download-progress/files/5276483/nativescript-download-progress-2.0.0.tgz.zip)

Zip containing a tgz for this PR for those who need NS7 support before merge and publish.

**Comment by toby-wollit** [comment]
Hi thank you for this - much appreciated. Could you please remove the prettier config and use the existing eslint config for formatting and ill merge it in? If you don't have time I'll tackle it when I have time.

Thanks again.

**Comment by tobydeh** [comment]
Hi thank you for this - much appreciated. Could you please remove the prettier config and use the existing eslint config for formatting and ill merge it in? If you don't have time I'll tackle it when I have time.

Thanks again.

**Comment by bradmartin** [comment]
I'll try to get back to that in a bit, didn't mean to push it up. Was trying to keep it consistent and my setup just overrides eslint configs 🙃 so I added that for my dev to avoid some silly file changes, but yep I'll get it cleaned up in a bit.

**Comment by bradmartin** [comment]
@tobydeh - apologies but I'm not sure how to use eslint config for formatting, not with VSCode anyway. When formatting it's using double quotes which is not what the source originally had so it's not being applied or maybe some rules locally are overwriting the config. 

If you have a command or flow to follow I'd be glad to do it though.

**Comment by bradmartin** [comment]
Also just pushed a small change that I noticed where I wasn't added the correct weakref to the iOS delegate, should work now 👍

**Comment by tobydeh** [comment]
No problem - I'll sort it and publish a new version.

**Comment by bradmartin** [comment]
Be sure the ts-patch install runs before publishing, not sure what flow you follow but if it doesn't the `@NativeClass()` decorator on the delegate will end up outputting ES2017 for the extends class there and not ES5 which is what the runtimes need for native classes that are extended.

File: src/package.json [hunk 1]
```diff
@@ -15,7 +15,7 @@
     "url": "https://github.com/tobydeh/nativescript-download-progress.git"
   },
   "scripts": {
-    "tsc": "npm i && tsc",
+    "tsc": "npm i && ts-patch install && tsc --skipLibCheck",
```
Comment (bradmartin) [inline src/package.json:18]: This ts-patch install is critical for NS7 support

**Comment by tobydeh** [comment]
1.4.0 published

**Comment by bradmartin** [comment]
Probably should have bumped a major with this one @tobydeh since it's a breaking change for NS and won't work with users on NS6 who have `^1.3.0` in their dependencies.

**Pull request closed without merging**